packages/victory-core/src/victory-util/collection.tsx

```diff
@@ -40,15 +38,23 @@ export function removeUndefined(arr) {
   return arr.filter((el) => el !== undefined);
 }
 
-export function getMaxValue(arr, ...values) {
-  const array = arr.concat(values);
+export function getMaxValue<T extends number | Date>(
+  arr: T[],
+  ...values: T[]
+): T {
+  const array: any[] = arr.concat(values);
+  // @ts-expect-error "Date is not assignable to number"
   return containsDates(array)
```

Comment:
I was attempting to make it generic, so that `getMaxValue([1,2,3]) => number` while `getMaxValue([new Date(), ...]) => Date`.  However, I'm not even sure if that's useful ... it's probable that the "consumer" of this function doesn't know if it's `number | Date` so they probably expect the result to be `number | Date`.

I'll move forward with your suggestion!

Reply:
Done. Removed the generics.  Instead of using `.map(...)` I just did ` as number[]` and left a comment about date coercion.